platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_tasks.adoc

```diff
@@ -130,3 +130,59 @@ include::sample[dir="snippets/bestPractices/useCacheabilityAnnotations/groovy",f
 === Tags
 
 `<<tags_reference.adoc#tag:tasks,#tasks>>`, `<<tags_reference.adoc#tag:caching,#caching>>`
+
+[[avoid_provider_get_outside_task_action]]
+== Do not call `get()`` on a Provider outside of a Task action
+:keywords: tasks,inputs-and-outputs
+
+When configuring tasks and extensions do not call link:{javadocPath}/org/gradle/api/provider/Provider.html#get()[`get()`] on a provider, use link:{javadocPath}/org/gradle/api/provider/Provider.html#map(org.gradle.api.Transformer)[`map()`], or link:{javadocPath}/https://docs.gradle.org/nightly/javadoc/org/gradle/api/provider/Provider.html#flatMap(org.gradle.api.Transformer)[`flatMap()`] instead
```

Comment:
🤔 Is there something to be added here about this being an ideal, as some tasks (especially on 3rd party plugins) might not use lazy properties and therefor this might not always be possible?

Reply:
I would argue that you should never directly interact with tasks created by 3rd party plugins. Maybe we should have a separate best practice on that.

Additionally, if someone has no provider, then they are not going to this practice anyway.